### DISCUSSION TITLE
config/component

### DISCUSSION THREAD

**Discussion by giscus** [ATQQ/sugar-blog#98] [Announcements]
# config/component

详细介绍主题提供的组件能力

https://theme.sugarat.top/config/component.html

**Comment by lpdswing** [comment]
文章开头的tab效果貌似build之后才正常, dev模式不行

**Reply by ATQQ** [reply]
嗯？ 就这个tabs吗 还是指哪个tab呀

**Reply by ATQQ** [reply]
我这复现不了，你看你那边能稳定复现么？

![](https://img.cdn.sugarat.top/mdImg/MTY4NjQ3MjMxNjE1OQ==686472316159)

**Comment by peigoz** [comment]
有开放自定义组件的能力吗？最近想开发个自己用的组件

**Reply by ATQQ** [reply]
按照Vite的插件规范弄就行，常规的就两种
1. 利用插槽将你的组件渲染进去，这是我自己拆的 一些 https://theme.sugarat.top/plugins/   

有个模板你可以看这个里面的原理介绍 https://sugarat.top/technology/works/vitepress-plugin-announcement.html#%E6%9C%80%E5%90%8E

模板 https://github.com/ATQQ/sugar-blog/tree/master/template/vitepress-plugin-slot-inject-template

2. 就是markdown-it 插件 渲染自己的语法


这些丢给AI 现在 一会儿就出来了

**Reply by peigoz** [reply]
> 按照Vite的插件规范弄就行，常规的就两种
> 
> 1. 利用插槽将你的组件渲染进去，这是我自己拆的 一些 https://theme.sugarat.top/plugins/
> 
> 有个模板你可以看这个里面的原理介绍 https://sugarat.top/technology/works/vitepress-plugin-announcement.html#%E6%9C%80%E5%90%8E
> 
> 模板 https://github.com/ATQQ/sugar-blog/tree/master/template/vitepress-plugin-slot-inject-template
> 
> 2. 就是markdown-it 插件 渲染自己的语法
> 
> 这些丢给AI 现在 一会儿就出来了

佬，我的意思是想在markdown用xmind-embed-viewer写一个支持xmind文件渲染的组件。我看了下需要在enhanceApp钩子中用app.component挂载。不过貌似没开放这个挂载自定义全局组件的能力

**Reply by ATQQ** [reply]
有的吧 和官方的一样，你用这个主题的话 没区别

https://theme.sugarat.top/config/style.html#%E6%A0%B7%E5%BC%8F%E9%85%8D%E7%BD%AE

https://vitepress.dev/zh/guide/extending-default-theme#registering-global-components

**Reply by ATQQ** [reply]
<img width="927" height="502" alt="image" src="https://github.com/user-attachments/assets/825e1e26-5d30-441e-96d9-ae8de0b519bb" />

**Reply by peigoz** [reply]
> https://vitepress.dev/zh/guide/extending-default-theme#registering-global-components

我在这里重新写一个enhanceApp会不会覆盖你主题的行为。我看你内部有注册一些东西的。。我还挺喜欢你主题的一些内置插件的。不想覆盖你主题的前提下给我自己注册一些自己用的全局组件就行了。

**Comment by peigoz** [comment]
试了下解决了。
```
export default {
  ...BlogTheme,
  enhanceApp(ctx) {
    // 保留主题自带的 enhanceApp 逻辑（注册 UserWorksPage、ProductCard 等）
    BlogTheme.enhanceApp?.(ctx)
    // 注册 XMind 思维导图嵌入组件，文章中通过 <XmindViewer url="..." /> 使用
    ctx.app.component('XmindViewer', XmindViewer)
  },
}
```

**Reply by ATQQ** [reply]
可以的 我在文档里也补充一下
